google/cloud/aiplatform/training_jobs.py

```diff
@@ -5335,6 +5335,100 @@ def run(
         )
 
 
+class TemporalFusionTransformerForecastingTrainingJob(_ForecastingTrainingJob):
+    _model_type = "TFT"
+    _training_task_definition = schema.training_job.definition.tft_forecasting
+    _supported_training_schemas = (schema.training_job.definition.tft_forecasting,)
+
+    def run(
+        self,
+        dataset: datasets.TimeSeriesDataset,
+        target_column: str,
+        time_column: str,
+        time_series_identifier_column: str,
+        unavailable_at_forecast_columns: List[str],
+        available_at_forecast_columns: List[str],
+        forecast_horizon: int,
+        data_granularity_unit: str,
+        data_granularity_count: int,
+        training_fraction_split: Optional[float] = None,
+        validation_fraction_split: Optional[float] = None,
+        test_fraction_split: Optional[float] = None,
+        predefined_split_column_name: Optional[str] = None,
+        timestamp_split_column_name: Optional[str] = None,
+        weight_column: Optional[str] = None,
+        time_series_attribute_columns: Optional[List[str]] = None,
+        context_window: Optional[int] = None,
+        export_evaluated_data_items: bool = False,
+        export_evaluated_data_items_bigquery_destination_uri: Optional[str] = None,
+        export_evaluated_data_items_override_destination: bool = False,
+        quantiles: Optional[List[float]] = None,
+        validation_options: Optional[str] = None,
+        budget_milli_node_hours: int = 1000,
+        model_display_name: Optional[str] = None,
+        model_labels: Optional[Dict[str, str]] = None,
+        model_id: Optional[str] = None,
+        parent_model: Optional[str] = None,
+        is_default_version: Optional[bool] = True,
+        model_version_aliases: Optional[Sequence[str]] = None,
+        model_version_description: Optional[str] = None,
+        additional_experiments: Optional[List[str]] = None,
+        hierarchy_group_columns: Optional[List[str]] = None,
+        hierarchy_group_total_weight: Optional[float] = None,
+        hierarchy_temporal_total_weight: Optional[float] = None,
+        hierarchy_group_temporal_total_weight: Optional[float] = None,
+        window_column: Optional[str] = None,
+        window_stride_length: Optional[int] = None,
+        window_max_count: Optional[int] = None,
+        holiday_regions: Optional[List[str]] = None,
+        sync: bool = True,
+        create_request_timeout: Optional[float] = None,
+    ) -> models.Model:
```

Comment:
Docstring. What's the difference between this and the superclass `run`? If there is none, I think this can be removed.

Reply:
Thanks Sasha. For some background, I based this PR on the [Seq2SeqPlus PR](https://github.com/googleapis/python-aiplatform/pull/1196) since both are adding a new forecasting model.

It looks like in that PR the decision was made to keep `run` around to force generation of docs for this method. Along the same lines, I believe the class documentation from the `_ForecastingTrainingJob` base class' docstring is sufficient since all the changes between these two classes is the model schema being sent in the training request.

I think it's important that we maintain consistency between the `Seq2SeqPlusForecastingTrainingJob`, `AutoMLForecastingTrainingJob`, and this new TFT variant. Based on that previous PR's context, do you think it's okay to leave this as is?

Reply:
Sounds good. I opened to ticket to add inherited members to the docs. [b/249069357](b/249069357).

Please add a TODO comment referencing this ticket.

Reply:
Thanks Sasha. I see the bug has been closed. I'll get rid of `run` for this as well.